.github/workflows/update-public-docs.yml

```diff
@@ -59,31 +62,90 @@ jobs:
           prerelease: false
 
       - name: Upload Release Asset
-        if: ${{ steps.check_docs.outputs.DOCS_CHANGED }}
-        id: upload-release-asset
+        if: ${{ steps.check_release_changed.outputs.DOCS_CHANGED }}
+        id: upload_release_asset
         uses: actions/upload-release-asset@v1
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          upload_url: ${{ steps.create-release.outputs.upload_url }}
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
           asset_path: ./new_public_docs.json
           asset_name: public_docs.json
           asset_content_type: application/json
 
-      - name: Trigger update in docs repo
-        if: ${{ steps.check_docs.outputs.DOCS_CHANGED }}
+      - name: Clone docs repo
+        if: ${{ steps.check_release_changed.outputs.DOCS_CHANGED }}
+        uses: actions/checkout@v3
+        with:
+          repository: opensafely/documentation
+          path: docs
+          ref: 'main'
+          token: ${{ secrets.DOCS_WRITE_TOKEN }}
+
+      - name: Check if update branch exists
+        if: ${{ steps.check_release_changed.outputs.DOCS_CHANGED }}
+        id: check-branch
+        run: |
+          cd docs
+          echo ::set-output name=branch-exists::`git ls-remote --heads origin ${BRANCH}`
+
+      - name: Create or checkout branch
+        if: ${{ steps.check_release_changed.outputs.DOCS_CHANGED }}
+        run: |
+          cd docs
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          if [ -n "${{ steps.check-branch.outputs.branch-exists }}" ]; then
+            echo "Checking out existing branch"
+            git fetch origin $BRANCH
+            git checkout $BRANCH
+          else
+            echo "Creating new branch"
+            git checkout -b $BRANCH
+          fi
+
+      - name: Commit new docs file
+        if: ${{ steps.check_release_changed.outputs.DOCS_CHANGED }}
+        run: |
+          echo ${{ steps.check_docs.outputs.DOCS_CHANGED }}
+          mv new_public_docs.json docs/public_docs.json
+          cd docs
+          git add public_docs.json
+          git commit -m "Update databuilder docs"
+          git push origin $BRANCH
+
+      # Open a PR for this branch, if one isn't already open
+      - name: Open PR
+        if: ${{ steps.check_release_changed.outputs.DOCS_CHANGED }}
+        id: list-pr
```

Comment:
question: Are we using this `id` anywhere?

Reply:
No - it's a leftover from a previous version of the workflow